manticore/core/cpu/disasm.py

```diff
@@ -0,0 +1,281 @@
+        else:
+            self.current_func = blocks[0].function
+        ##################
+
+        il = self._pop_from_il_queue(code, pc)
+        self.insn_size = il.size
+        self.current_pc = pc
+        self.disasm_il = il
+
+        # create an instruction from the fallback disassembler if Binja can't
+        o = il.operation
+        if (o == enums.LowLevelILOperation.LLIL_UNIMPL or
+                o == enums.LowLevelILOperation.LLIL_UNIMPL_MEM):
+            return self.fallback_disasm.disassemble_instruction(code, pc)
+
+        return self.BinjaILInstruction(il,
+                                       self.entry_point_diff,
+                                       self.disasm_insn_size,
+                                       self.current_llil_func)
+
+
+    class BinjaILInstruction(Instruction):
+        def __init__(self, llil, offset, size, function):
+            self.llil = llil
+            self.offset = offset
+            self._size = size
+            self.function = function
+            super(BinjaILDisasm.BinjaILInstruction, self).__init__()
+
+        def _fix_addr(self, addr):
+            return addr + self.offset
+
+        @property
+        def size(self):
+            return self._size
+
+        @property
+        def mnemonic(self):
+            return self.llil.operation.name
+
+        @property
+        def op_str(self):
+            return " ".join([str(x.op) for x in self.llil.operands])
+
+        @property
+        def operands(self):
+            return self.llil.operands
+
+        @operands.setter
+        def operands(self, value):
+            # This will be overloaded by a BinjaILOperand
+            self.llil.operands = value
+
+        @property
+        def insn_name(self):
+            return self.llil.operation.name
+
+        @property
+        def name(self):
+            return self.llil.operation.name[len("LLIL_"):]
+
+        @property
+        def address(self):
+            return self._fix_addr(self.llil.address)
+
+        @property
+        def sets_pc(self):
+            import binaryninja.enums as enums
+            op = self.llil.operation
+            return (op == enums.LowLevelILOperation.LLIL_CALL or
+                    op == enums.LowLevelILOperation.LLIL_JUMP or
+                    op == enums.LowLevelILOperation.LLIL_JUMP_TO or
+                    op == enums.LowLevelILOperation.LLIL_IF or
+                    op == enums.LowLevelILOperation.LLIL_RET or
+                    op == enums.LowLevelILOperation.LLIL_SYSCALL or
+                    op == enums.LowLevelILOperation.LLIL_GOTO)
+
+        def __repr__(self):
+            return ("%d %s\t%s %s %x") % (self.llil.instr_index,
+                                          hex(self.llil.address),
+                                          str(self.llil),
+                                          self.llil.operation.name,
+                                          self.llil.address)
+
+class BinjaDisasm(Disasm):
```

Comment:
this is unused right? since we removed support for just binja disasm in `init_disassembler` can we remove it here also?